File: go/main4.go
package main
import "fmt"
func dangerous(value1 int, value2 int) int { return value1 / value2 }
func foo(array []int, counter int) int {
    if counter == 0 { return dangerous(array[0], counter) }
    return foo(array, counter-1)
}
func main(){ array := make([]int, 1000); fmt.Printf("The result is %d\n", foo(array, 6)) }

File: ocaml/main4.ml
let dangerous v1 v2 = v1 / v2
let rec foo array counter =
  if counter = 0 then dangerous array.(0) counter else foo array (counter - 1)
let () =
  let array = Array.make 1000 0 in
  Printf.printf "The result is %d\n" (foo array 6)

